### PR TITLE
Fix API proxy race condition

### PR DESCRIPTION
**Pull request by huguesBouvier**
Fixes issue 9653726
At start up API proxy waits for the trust bundle and server cert before starting. 
However if we get them too close together, it will receive only one event.
 
There is a "debouncing" of event made by the signaling system:
`Notify provides two APIs. The default API debounces events (if the backend reports two similar events in close succession, Notify will only report one).`

2 possible fixes:
1. Easy fix: remove debouncing. One liner. Though I am not sure of the impact. 
2. Change a bit the starting strategy, wait the have both cert before sending 1 signal.

This PR implements strategy 2, because the fix only impact start up. Once API proxy is running code path is same as before.

Test:
Stop and restarted API proxy 150 times.

### CODE REVIEW

File: edge-modules/api-proxy-module/src/main.rs [hunk 1]
```diff
@@ -106,10 +106,7 @@ pub fn nginx_controller_start(
         //Wait for configuration to be ready.
         notify_config_reload_api_proxy.notified().await;
 
-        //Wait for the trust bundle.
-        notify_cert_reload_api_proxy.notified().await;
-
-        //Wait for the server cert and private key.
+        //Wait for the trust bundle and server cert
```
Comment (varunpuranik) [inline edge-modules/api-proxy-module/src/main.rs:109]: Can this fix be simplified by using different instances of Notify for trust_bundle and server_cert?

**Comment by varunpuranik** [comment]
Can this fix be simplified by using 2 different instances of Notify for server certs and trust bundle? 
Otherwise this change looks good.

**Review by varunpuranik — DISMISSED** [review]

File: edge-modules/api-proxy-module/src/monitors/certs_monitor.rs [hunk 2]
```diff
@@ -128,6 +118,25 @@ pub fn start(
     Ok((monitor_loop, shutdown_handle))
 }
 
+async fn check_server_cert(cert_monitor: &mut CertificateMonitor) -> Result<bool, Error> {
```
Comment (nyanzebra) [inline edge-modules/api-proxy-module/src/monitors/certs_monitor.rs:121]: Since this is getting pulled out into a separate method should we add a UT?

Comment (huguesBouvier) [reply edge-modules/api-proxy-module/src/monitors/certs_monitor.rs:121]: Possible but would need to inject the method "file::write_binary_to_file"

**Comment by huguesBouvier** [comment]
> Can this fix be simplified by using 2 different instances of Notify for server certs and trust bundle?
> Otherwise this change looks good.

It can. I think you're right, it looks cleaner. Might as well do it. It is less changes so less risky

**Review by vadim-kovalyov — APPROVED** [review]

**Review by varunpuranik — APPROVED** [review]

**Pull request merged**